src/Stryker.Core/Stryker.Core/Initialisation/InputFileResolver.cs

```diff
@@ -46,23 +46,37 @@ public InputFileResolver(IFileSystem fileSystem, IProjectFileReader projectFileR
         /// </summary>
         public ProjectInfo ResolveInput(StrykerOptions options)
         {
-            var result = new ProjectInfo();
-            var projectFile = ScanProjectFile(options.BasePath);
+            var projectAnalyzerResult = new ProjectInfo();
+            var testProjectFile = ScanProjectFile(options.BasePath);
 
             // Analyze the test project
-            result.TestProjectAnalyzerResult = _projectFileReader.AnalyzeProject(projectFile, options.SolutionPath);
+            projectAnalyzerResult.TestProjectAnalyzerResult = _projectFileReader.AnalyzeProject(testProjectFile, options.SolutionPath);
 
             // Determine project under test
             var reader = new ProjectFileReader();
-            var projectUnderTest = reader.DetermineProjectUnderTest(result.TestProjectAnalyzerResult.ProjectReferences, options.ProjectUnderTestNameFilter);
+            var projectUnderTest = reader.DetermineProjectUnderTest(projectAnalyzerResult.TestProjectAnalyzerResult.ProjectReferences, options.ProjectUnderTestNameFilter);
 
             // Analyze project under test
-            result.ProjectUnderTestAnalyzerResult = _projectFileReader.AnalyzeProject(projectUnderTest, options.SolutionPath);
+            projectAnalyzerResult.ProjectUnderTestAnalyzerResult = _projectFileReader.AnalyzeProject(projectUnderTest, options.SolutionPath);
+
+            var projectContents = new FolderComposite();
+            projectAnalyzerResult.FullFramework = !projectAnalyzerResult.TestProjectAnalyzerResult.TargetFramework.Contains("netcoreapp", StringComparison.InvariantCultureIgnoreCase);
+            var projectUnderTestDir = Path.GetDirectoryName(projectAnalyzerResult.ProjectUnderTestAnalyzerResult.ProjectFilePath);
+
+            // Search for compile linked source files in project under test
+            IDictionary<string, string> compileIncludeLinkedFiles = FindCompileLinkedFiles(projectAnalyzerResult.ProjectUnderTestAnalyzerResult.ProjectFilePath);
 
-            var inputFiles = new FolderComposite();
-            result.FullFramework = !result.TestProjectAnalyzerResult.TargetFramework.Contains("netcoreapp", StringComparison.InvariantCultureIgnoreCase);
-            var projectUnderTestDir = Path.GetDirectoryName(result.ProjectUnderTestAnalyzerResult.ProjectFilePath);
-            foreach (var dir in ExtractProjectFolders(result.ProjectUnderTestAnalyzerResult, result.FullFramework))
+            var sharedProjectFilePaths = FindSharedProjects(projectAnalyzerResult.ProjectUnderTestAnalyzerResult);
+
+            var projectFolders = ExtractProjectFolders(projectAnalyzerResult.ProjectUnderTestAnalyzerResult, sharedProjectFilePaths);
+
+            // Search for compile linked source files in all linked shared projects
+            foreach (var sharedProjectFile in sharedProjectFilePaths)
+            {
+                compileIncludeLinkedFiles.ToList().AddRange(FindCompileLinkedFiles(sharedProjectFile));
```

Comment:
`AddRange` is invoked on a newly created `List`, so `compileIncludeLinkedFiles` is never updated with the results from `FindCompileLinkedFiles`.

Reply:
Oh wow, thanks! Fixed.